feat: Single Model handler with OAuth 2.0 Client Credentials support

kkedziak-splunk opened the pull request:
**Issue number:**

### PR Type

**What kind of change does this PR introduce?**
* [ ] Feature
* [ ] Bug Fix
* [ ] Refactoring (no functional or API changes)
* [ ] Documentation Update
* [ ] Maintenance (dependency updates, CI, etc.)

## Summary

### Changes

Please provide a summary of the changes.

### User experience

Please describe the user experience before and after this change. Screenshots are welcome for additional context.

## Checklist

If an item doesn't apply to your changes, leave it unchecked.

### Review

* [ ] self-review - I have performed a self-review of this change according to the [development guidelines](https://splunk.github.io/addonfactory-ucc-generator/contributing/#development-guidelines)
* [ ] Changes are documented. The documentation is understandable, examples work [(more info)](https://splunk.github.io/addonfactory-ucc-generator/contributing/#documentation-guidelines)
* [ ] PR title and description follows the [contributing principles](https://splunk.github.io/addonfactory-ucc-generator/contributing/#pull-requests)
* [ ] meeting - I have scheduled a meeting or recorded a demo to explain these changes (if there is a video, put a link below and in the ticket)

### Tests

See [the testing doc](https://splunk.github.io/addonfactory-ucc-generator/contributing/#build-and-test).

* [ ] Unit - tests have been added/modified to cover the changes
* [ ] Smoke - tests have been added/modified to cover the changes
* [ ] UI - tests have been added/modified to cover the changes
* [ ] coverage - I have checked the code coverage of my changes [(see more)](https://splunk.github.io/addonfactory-ucc-generator/contributing/#checking-the-code-coverage)

**Demo/meeting:**

*Reviewers are encouraged to request meetings or demos if any part of the change is unclear*

soleksy-splunk commented on tests/testdata/expected_addons/expected_output_global_config_everything/Splunk_TA_UCCExample/bin/splunk_ta_uccexample_rh_account.py:
```diff
@@ -271,10 +274,74 @@
+        client_id = client_id or self.callerArgs.data.get("client_id", [None])[0]
+
+        client_secret = self.callerArgs.data.get("client_secret_oauth_credentials", [None])[0]
+        client_secret = client_secret or self.callerArgs.data.get("client_secret", [None])[0]
```
is it handler just for client credentials or also for `Authorization Code` (current oauth)? just wondering if there can't be a mixup when someone sets up `client_id_oauth_credentials` and  `client_secret`, without other params.

If thats intended to use of those params thats fine and please ignore :D

kkedziak-splunk replied:
It was added in case someone specifies alternative names but I can remove that, as I already wrote in the documentation that we require `*_oauth_credentials` fields.

kkedziak-splunk commented on splunk_add_on_ucc_framework/commands/rest_builder/endpoint/single_model.py:
```diff
@@ -83,8 +85,113 @@ def generate_rh(self) -> str:
+        )
+    )
+    _cls_template = """
+APP_NAME = import_declare_test.ta_name
```
I didn't know that. I will change it.

kkedziak-splunk commented:
@sgoral-splunk I addressed the comments and also added a small change:

https://github.com/splunk/addonfactory-ucc-generator/pull/1715/files#diff-4c0923e6773b00c1916127e8c8e873a15e04a6bb9b01addd52b98a0a7fad8dc0R189-R192

To not add this "if" for `auth_type` when only `oauth_client_credentials` is specified, as for only one type we won't have that field in the model.
